carriage_cargo.rb

```diff
@@ -1,5 +1,23 @@
 class CarriageCargo < Carriage
-  def initialize
+
+  def initialize(volume)
+    @volume = volume
+    @holden_volume = 0
     @type = :cargo
   end
+
+  def take_volume(volume)
+    if volume <= @volume
+      @volume -= volume and @holden_volume += volume
+    end
+    
+  end
+
+  def volume
```

Comment:
```suggestion
  def holden_volume
```
тогда уж так, чтобы не путаться?

Reply:
и еще важный момент
обрати внимание на схожесть кода в обоих подклассах класса Вагон
если подумать, то большую часть этого кода можно вынести в родительский класс